pkg/operator/datasources.go

```diff
@@ -230,6 +230,7 @@ func (op *Reporting) handlePrometheusMetricsDataSource(logger log.FieldLogger, d
 	// run the import
 	results, err := importer.ImportFromLastTimestamp(context.Background())
 	if err != nil {
+		op.eventRecorder.Event(dataSource, v1.EventTypeWarning, "FailedPrometheusQuery", "Unable to import metrics after Prometheus query failure. Check the reporting-operator container logs for more information.")
```

Comment:
We may want to expose a configuration for disabling this kind of check as this could be potentially expensive if every ReportDataSource promQL query is constantly failing.

Reply:
Well maybe expensive at first, but that should normalize over time as we're not going to be updating these events as there's going to be no ReportDataSource events to process in the cache as we're constantly failing to import metrics, at least in this scenario.